packages/table/src/table.tsx

```diff
@@ -489,10 +489,26 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         );
     }
 
-    public resizeRowsByTallestCell(columnIndex: number) {
+    /**
+     * Resize all rows in the table to the height of the tallest visible cell in the specified columns.
+     * If no indices are provided, default to using the tallest visible cell from all columns in view.
+     */
+    public resizeRowsByTallestCell(columnIndices?: number | number[]) {
         const { locator } = this.state;
-
-        const tallest = locator.getTallestVisibleCellInColumn(columnIndex);
+        let tallest = 0;
+        if (columnIndices == null) {
+            // Consider all columns currently in viewport
+            const { grid } = this;
+            const { viewportRect } = this.state;
```

Comment:
i prefer not destructuring variables that are used exactly once. can remove two lines by simply using `this.grid` once and `this.state.viewportRect` once (in the same line 😮 )